### PR TITLE
rename BatchNormGradFunctor

### PR DESCRIPTION
**Pull request by zhangyuqin1998**
### PR types

Others 
### PR changes

Others 
### Description

将Kernel删除改为Function

### CODE REVIEW

File: paddle/phi/kernels/funcs/batch_norm_grad_functors.cu.h
Comment (zyfncg) [inline paddle/phi/kernels/funcs/batch_norm_grad_functors.cu.h:1]: 这两个头文件的代码有些复杂，可以合并下头文件，尝试下把实现放到.cc和.cu里

File: paddle/phi/kernels/funcs/batch_norm_grad_functors.h
Comment (zyfncg) [inline paddle/phi/kernels/funcs/batch_norm_grad_functors.h:1]: same

**Review by zyfncg — DISMISSED** [review]

**Review by zyfncg — APPROVED** [review]
